tests/functional/ovndbcluster_controller_test.go

```diff
@@ -298,6 +328,119 @@ var _ = Describe("OVNDBCluster controller", func() {
 			}).Should(Succeed())
 		})
 
+		It("should create an external ConfigMap with expected key-value pairs and OwnerReferences set", func() {
+			internalAPINADName := types.NamespacedName{Namespace: namespace, Name: "internalapi"}
+			nad := th.CreateNetworkAttachmentDefinition(internalAPINADName)
+			DeferCleanup(th.DeleteInstance, nad)
+
+			statefulSetName := types.NamespacedName{
+				Namespace: namespace,
+				Name:      "ovsdbserver-sb",
+			}
+			th.SimulateStatefulSetReplicaReadyWithPods(
+				statefulSetName,
+				map[string][]string{namespace + "/internalapi": {"10.0.0.1"}},
+			)
+
+			externalCM := types.NamespacedName{
+				Namespace: OVNDBClusterName.Namespace,
+				Name:      "ovncontroller-config",
+			}
+
+			ExpectedExternalSBEndpoint := "tcp:ovsdbserver-sb." + namespace + ".svc:6642"
+
+			Eventually(func() corev1.ConfigMap {
+				return *th.GetConfigMap(externalCM)
+			}, timeout, interval).ShouldNot(BeNil())
+
+			// Check OwnerReferences set correctly for the Config Map
+			Expect(th.GetConfigMap(externalCM).ObjectMeta.OwnerReferences[0].Name).To(Equal(OVNDBClusterName.Name))
+			Expect(th.GetConfigMap(externalCM).ObjectMeta.OwnerReferences[0].Kind).To(Equal("OVNDBCluster"))
+
+			Eventually(func(g Gomega) {
+				g.Expect(th.GetConfigMap(externalCM).Data["ovsdb-config"]).Should(
+					ContainSubstring("ovn-remote: %s", ExpectedExternalSBEndpoint))
+			}, timeout, interval).Should(Succeed())
+		})
+
+		It("should create an external ConfigMap with ovn-encap-type if OVNController is configured", func() {
```

Comment:
please add a test case with OVNController named differently (not `ovncontroller`).

Reply:
maybe check that removal of OVNController resets the CM to not include encap type?

Reply:
Regarding creating a test case with OVNController named differently to ovncontroller, currently the tests creates an OVNController object called "name := "ovncontroller-" + uuid.New().String()"